### ISSUE TITLE
Failed building wheel in Python 3.10(beta1): "PyFrameObject has no member named 'f_stacktop'"

### ISSUE THREAD

**Issue by nchepanov** [jpype-project/jpype#978]
`JPype1==1.2.1`

```
   native/common/jp_exception.cpp:506:9: error: 'PyFrameObject {aka struct _frame}' has no member named 'f_executing'
     frame->f_executing = 0;
           ^~~~~~~~~~~
   native/common/jp_exception.cpp:515:9: error: 'PyFrameObject {aka struct _frame}' has no member named 'f_stacktop'; did you mean 'f_stackdepth'?
     frame->f_stacktop = NULL;
           ^~~~~~~~~~
            f_stackdepth
   error: command '.../bin/gcc' failed with exit code 1
Building wheel for JPype1 (PEP 517): finished with status 'error'
ERROR: Failed building wheel for JPype1
```

it looks like it was replaced:

```diff
- char f_executing;           /* whether the frame is still executing */    
+ PyFrameState f_state;       /* What state the frame is in */
```

see https://github.com/python/cpython/commit/cb9879b948a19c9434316f8ab6aba9c4601a8173 I can't find this in the Python documentation

**Comment by Thrameos** [comment]
They have not yet given an API to create a 3rd party stack frame and my organization has barred me from contributing to Python (because IPO won't sign the Python License agreement.)   This unfortunately means that we have to replicate Python API internals to convert Java frames into Python frames when transferring a Java exception to Python.  Python doesn't accept duck typing so it must be the real structure and not some object that had the correct fields when probed.

The solution is to do the following.
1) Add a ``#define`` for version 3.10 on skip the offending 2 lines code (which are just initializing memory so Python does not crash).
2) Read the source code to determine what ``f_state`` would be required for our stack frames.   (or documentation if it exists)
3) Add an ``#else`` clause with the correct code.

I suppose we can try to do a ``memset`` to clear the whole structure rather than trying to set all the fields manually, but that will lead to other issues in the case where a field needs to be no-zero.   That is why I deliberately made the code fail when they changed the structure.

**Comment by Thrameos** [comment]
Thanks for the bug report.  This was fixed by #982.
